File: summer/WebContent/js/system/windows/controls/ContentModelTreeEnumerator.js
/**
 * ContentModelTreeEnumerator
 */

define(["dojo/_base/declare", "system/Type", "controls/ModelTreeEnumerator"], function(declare, Type, ModelTreeEnumerator){
	var ContentModelTreeEnumerator = declare("ContentModelTreeEnumerator", ModelTreeEnumerator,{
   	 	"-chains-": {
	      constructor: "manual"
	    },
		constructor:function(/*ContentControl*/ contentControl, /*object*/ content) 
        { 
			ModelTreeEnumerator.prototype.constructor.call(this, content);
//            Debug.Assert(contentControl != null, "contentControl should be non-null."); 

            this._owner = contentControl; 
        }
	});
	
	Object.defineProperties(ContentModelTreeEnumerator.prototype,{
//        protected override bool 
		IsUnchanged:
        { 
            get:function()
            { 
                return Object.ReferenceEquals(this.Content, this._owner.Content); 
            }
        }
	});
	
	ContentModelTreeEnumerator.Type = new Type("ContentModelTreeEnumerator", 
			ContentModelTreeEnumerator, [ModelTreeEnumerator.Type]);
	return ContentModelTreeEnumerator;
});

File: org.summer.view.widget/src/org/summer/view/widget/controls/ContentModelTreeEnumerator.java
package org.summer.view.widget.controls;

/*internal*/ public class ContentModelTreeEnumerator extends ModelTreeEnumerator
{ 
    /*internal*/ public ContentModelTreeEnumerator(ContentControl contentControl, Object content) 
    { 
    	super(content)
//        Debug.Assert(contentControl != null, "contentControl should be non-null."); 

        _owner = contentControl; 
    }

    protected /*override*/ boolean IsUnchanged
    { 
        get
        { 
            return Object.ReferenceEquals(Content, _owner.Content); 
        }
    } 

    private ContentControl _owner;
}